src/map/clif.cpp

```diff
@@ -25204,6 +25219,25 @@ static int clif_parse(int fd)
 	}
 #endif
 
+	ARR_FIND(0,packet_spammer.size(),piter,packet_spammer[piter].client_addr == session[fd]->client_addr);
```

Comment:
on every processed packet, we do a linear search on a list of recent packet senders? Why not just put this inside `map_session_data`?

Also, this is a c++ vector, you can just use `std::find()`.
this also doesnt work on dual clients, only the first client will get kicked

Reply:
changed to store fd, it's solving that?